teps/0046-pipelinerun-in-a-pod.md

```diff
@@ -0,0 +1,97 @@
+schedule the different TaskRun Pods in an appropriate way, especially to allow multiple of them to run
+concurrently while accessing the same Workspace volume.
+
+[#3638: Investigate if we can run whole PipelineRun in a Pod](https://github.com/tektoncd/pipeline/issues/3638)
+
+[Design doc: Task parallelism when using workspace](https://docs.google.com/document/d/1lIqFP1c3apFwCPEqO0Bq9j-XCDH5XRmq8EhYV_BPe9Y)
+
+## Motivation
+
+Currently the only way to run a Pipeline composed of Tasks is that each task is executed in its own Pod. If you combine
+Tasks in a Pipeline and they need to share data (beyond a simple
+[result](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#using-results)),
+you'll need to provision a PVC or do some other similar, cloud specific thing,
+to [make a volume available](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md#specifying-volumesources-in-workspaces)
+that can be shared between them.
+
+Some problems with this:
+* PVCs add additional overhead, both in speed, in resource costs and in management.
+* PVCs is hard to share between multiple Pods concurrently - needs appropriate scheduling of the Pods.
+* PVCs is difficult to share in subsequently run Pods that might be scheduled to a different AZ than where the
+  volume is located.
+
+### Goals
+
+- Make it possible to run whole PipelineRun (including all TaskRuns) within a single Pod so that the workspace can be within the Pod:
+    - At runtime in a PipelineRun
+    - This should not change anything for the author of Tasks or Pipelines
+- No changes in the Pipeline or Task API for authors
+
+### Non-Goals
+
+- To fully handle the Pod resource requests and limits in an appropriate way for the PipelineRun Pod.
+
+### Use Cases (optional)
+
+- A user want to create a typical Build pipeline with git-clone, lint, code-build, unit-tests and multiple
+  parallel integration tests - all Tasks sharing the volume that was populated by git-clone and read by all
+  subsequent tasks to perform "validation". The user would like to see its PipelineRun run to completion as
+  a single unit INSTEAD OF worry about appropriately scheduled taskRun Pods in a busy cluster and need to
+  use Persistent Volumes for the files from git-clone.
+- A user want to run a Build pipeline that starts with two concurrent tasks, git-clone and another task that
+  fetches cached dependencies from a cache and then build the cloned code while reusing the cached dependencies
+  to speed up the build and see the full PipelineRun to be run in a single Pod, on a single Node WITHOUT NEEDING
+  to mount external volumes and worry that its Pipeline is composed of several Pods that might be scheduled to 
+  different cloud Availability Zones and reach a deadlocked state.
+- A user wants to create a Pipeline composed of Tasks from the catalog, to checkout code, run unit tests and upload results,
```

Comment:
This use-case is shared with [TEP-0044](https://github.com/tektoncd/community/pull/316) 👼🏼